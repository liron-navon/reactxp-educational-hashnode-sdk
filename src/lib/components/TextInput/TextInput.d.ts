/// <reference types="react" />
import { Types } from 'reactxp';
interface TextInputProps extends Types.TextInputProps {
    labelStyle?: Types.TextStyle;
    label?: string;
}
declare const TextInput: (props: TextInputProps) => JSX.Element;
export default TextInput;
