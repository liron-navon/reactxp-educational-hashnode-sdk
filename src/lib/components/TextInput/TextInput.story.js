import React from 'react';
import {storiesOf} from '@storybook/react';
import TextInput from './TextInput';
import {action} from '@storybook/addon-actions';

storiesOf('components/TextInput', module)
    .add('default', () => {
        return (
            <TextInput
                label='Add email label'
                accessibilityLabel={'Add email'}
                placeholder={'john@doe.com'}
                onChangeText={action('onChange')}
            />
        )
    });
