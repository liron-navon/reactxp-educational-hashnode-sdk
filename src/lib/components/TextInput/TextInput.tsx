import React from "react";
import {
    TextInput as RXTextInput,
    Types,
    View,
    Text
} from 'reactxp'
import theme from '../../theming/theme';
import {createStyleSheet} from '../../utils/styleHelpers';

// the prop types for the button
interface TextInputProps extends Types.TextInputProps {
    labelStyle?: Types.TextStyle,
    label?: string
}

const TextInput = (props: TextInputProps) => {
    const styles = createStyleSheet({
        textInput: {
            height: 40,
            padding: theme.spacing,
            ...theme.border
        },
        label: {
            fontSize: 14,
            paddingBottom: theme.spacing / 2
        }
    });

    return (
        <View>
            {
                props.label &&
                <Text style={[styles.label, props.labelStyle]}>
                    {props.label}
                </Text>
            }
            <RXTextInput
                {...props}
                style={[styles.textInput, props.style]}/>
        </View>
    )
};

export default TextInput;
