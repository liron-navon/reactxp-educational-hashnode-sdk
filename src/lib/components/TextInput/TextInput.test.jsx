import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

test('renders correctly', () => {
    const tree = renderer.create(<TextInput/>).toJSON();
    expect(tree).toMatchSnapshot();
});
