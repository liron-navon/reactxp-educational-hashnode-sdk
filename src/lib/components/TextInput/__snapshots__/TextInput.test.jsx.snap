// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  disabled={false}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onInput={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onPaste={[Function]}
  size={1}
  style={
    Object {
      "alignItems": "stretch",
      "borderColor": "grey",
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 0,
      "height": 40,
      "outline": "none",
      "overflowX": "hidden",
      "overflowY": "auto",
      "padding": 6,
      "position": "relative",
      "resize": "none",
      "width": "100%",
    }
  }
  type="text"
  value=""
/>
`;
